### DISCUSSION TITLE
Is there a list of Dagster reserved names/keys?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#8551] [Q&A]
Hey guys! Is there a place that specified the naming rules for Dagster ops/graphs/ins/outs/mapping keys? I’m assuming it’s the usual python rules with a few exceptions (like not using “context” or “output”), but it’ll be nice to see the entire list of rules/exceptions

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1641898015167900?thread_ts=1641898015.167900&cid=C01U954MEER).*

**Comment by yuhan** [comment]
It's not documented at the moment but you can find the list here in code: https://github.com/dagster-io/dagster/blob/0.15.0/python_modules/dagster/dagster/core/definitions/utils.py#L17-L42 (as of 0.15.0)
